app/src/main/java/chat/rocket/android/activity/ServerConfigActivity.java

```diff
@@ -77,14 +83,27 @@ protected void onPause() {
   }
 
   private void onRenderServerConfigError(ServerConfig config) {
-    if (config != null) {
+    if (config.getState() == ServerConfig.STATE_CONNECTION_ERROR) {
       sessionObserver.unsub();
       showFragment(new RetryConnectFragment());
     } else {
       sessionObserver.sub();
     }
   }
 
+  private void continueWithServerConfigSession(final Session session) {
+    fetchPublicSettings()
+        .continueWith(task -> {
+          registerForPush();
+          return task;
+        })
+        .continueWith(task -> {
+          onRenderServerConfigSession(session);
+          return task;
+        })
+        .continueWith(new LogcatIfError());
+  }
```

Comment:
Public settings contain some settings that may (will) be used by the app and may be applied before the login.

And by registering for push before the login we keep up with the current behavior already in place in the Cordova app and which could be specially useful on the white labels.